x-pack/plugins/apm/public/components/app/storage_explorer/index.tsx

```diff
@@ -67,11 +106,94 @@ export function StorageExplorer() {
         </EuiFlexItem>
       </EuiFlexGroup>
       <EuiSpacer />
+
+      {!calloutDismissed.optimizePerformance && (
+        <EuiCallOut
+          title={i18n.translate(
+            'xpack.apm.storageExplorer.longLoadingTimeCalloutTitle',
+            {
+              defaultMessage: 'Long loading time?',
+            }
+          )}
+          iconType="timeRefresh"
+        >
+          <p>
+            <FormattedMessage
+              id="xpack.apm.storageExplorer.longLoadingTimeCalloutText"
+              defaultMessage="Enable progressive loading of data or optimized sorting for services list in {kibanaAdvancedSettingsLink}."
+              values={{
+                kibanaAdvancedSettingsLink: (
+                  <EuiLink href={getKibanaAdvancedSettingsHref(core)}>
+                    {i18n.translate(
+                      'xpack.apm.storageExplorer.longLoadingTimeCalloutLink',
+                      {
+                        defaultMessage: 'Kibana advanced settings',
+                      }
+                    )}
+                  </EuiLink>
+                ),
+              }}
```

Comment:
I had it linked to APM general settings at first because `observability:apmProgressiveLoading` was there. But now both this and `observability:apmServiceInventoryOptimizedSorting` will only be available in the labs. We could open labs instead in the next iteration.